### PR TITLE
chore: switching from oclif to yargs

### LINKED ISSUE #325: Switch from oclif to yargs
oclif has been causing all sorts of oddities and weirdness, including a recent issue where it starts moaning about globby for no reason. 

```
$ spectral --version
(node:30001) [MODULE_NOT_FOUND] Error Plugin: @stoplight/spectral: Cannot find module 'globby'
module: @oclif/config@1.13.0
task: not loading commands, globby not found
plugin: @stoplight/spectral
root: /Users/phil/.asdf/installs/nodejs/10.15.3/.npm/lib/node_modules/@stoplight/spectral
See more details with DEBUG=*
@stoplight/spectral/4.0.0-beta1 darwin-x64 node-v10.15.3
```
Switch it out for Yargs as Prism has done. 

**Acceptance Criteria**
- No breaking changes to the CLI (e.g: its ok if help page looks a bit different)
- Oclif is gone
- Lint command is still tested, coverage going up would be nice

### PR DESCRIPTION
**Pull request by philsturgeon**
Fixes #325

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by XVincentX** [comment]
@philsturgeon Maybe put me as a reviewer when you're done — I made a BIG mistake while switching to Yargs that I do not remember, but I want to make sure we do not make it again with Spectral

**Comment by P0lip** [comment]
@philsturgeon @XVincentX 
I'm afraid we won't be able to implement the yargs testing approach that established in Prism.
There are a few significant differences between the case we deal with here and Spectral.
First of all,
- we don't need middleware, which seems to be needed to make `commandPromise` an actual promise,
- there is no way to grab output - we could mock `writeOutput`, yet this has a huge downside - it's not equal to actual output printed in the console, so we won't be able to test `--format=json` flag
- it does not wait until handler promise gets resolved, so there is no way to make an assertion when spectral has in fact finished linting

Prism does not face any of above.

That said, I suppose we'd need to resort to process spawning - I used this approach on a number of occasions and it's by far the most reliable.

Basically, we need something more or less like this
```js
import { promisify  } from 'util'

const exec = promisify(require('child_process').exec);

it('writes something', async () => {
  await { stdout, stderr} = exec('./bin/run lint foo.yaml');
  expect(stdout).toContain('blabala');
});
```

**Comment by philsturgeon** [comment]
Before trying the commandPromise.then approach I was using code from the yargs advanced documentation showing how you capture output... so why not? We are providing the help() format is output, so how is that output differently to what our commands are doing? Should we log somewhere else other than console?

If that approach cannot be saved any way, we can move writeOutput to the service, mock it, and just unit test what is being passed in. We then have unit tests on the service showing those inputs give the expected functionality? The integration will be skipped and end to end testing will cover it properly when we port a test harness (#304).

**Comment by P0lip** [comment]
> If that approach cannot be saved any way, we can move writeOutput to the service, mock it, and just unit test what is being passed 

I tried that, but it gets called too late. commandPromise promise does not wait until handler is resolved.
We'd need to the whole handler to the linter service and mock it.

As said, that approach won't let us catch `console.warn` calls and such, but if we mean to have unit tests for now, we should be okay.

**Comment by philsturgeon** [comment]
I still don't quite understand. I don't care about commandPromise, that was just me trying something out.

https://github.com/yargs/yargs/blob/master/docs/advanced.md

```
it("returns help output", async () => {
  // Initialize parser using the command module
  const parser = yargs.command(require('./my-command-module')).help();

  // Run the command module with --help as argument
  const output = await new Promise((resolve) => {
    parser.parse("--help", (err, argv, output) => {
      resolve(output);
    })
  });

  // Verify the output is correct
  expect(output).toBe(expect.stringContaining("helpful message"));
});
```

1. Why can we test help like this (this works) but we cannot test anything other than help? 
2. If we know the difference, can we can make our code log more like help?

**Comment by P0lip** [comment]
@philsturgeon 
Uhhh, now I get what you mean.
Let me see

**Comment by philsturgeon** [comment]
@P0lip ok, thats a bit of a shame but also your code is looking amazing in this branch now. Do you think you a can take this over the finishing line with some unit tests on the services, and any sort of test which shows the CLI is doing something productive, and I'll get to work on #368 based off of this branch?

**Comment by P0lip** [comment]
@philsturgeon 
The CI might be green at this point, yet I'd like to enable a few more tests in `src/cli/services/__tests__/linter.test.ts`

**Comment by P0lip** [comment]
Although, we all want to get rid of oclif so badly I'd suggest holding on with this one.
I don't feel confident enough to to tell it does not introduce any breaking changes.
We don't verify stdout too much now, therefore in particular json format might (or might not) be broken.
If someone can test a few cases manually to verify the actual output printed to stdout, this should be safe to go. 

Other than that, should be good to review! @philsturgeon, @marbemac, @XVincentX it's all yours now.

**Comment by philsturgeon** [comment]
Ok, seems like we need the test harness before this one can be done. I'll get started on #304 now.

**Comment by P0lip** [comment]
`--format=json` is broken since we log `OpenAPI 3.x detected` now.
I'll take care of it.

**Comment by philsturgeon** [comment]
@XVincentX ready! :D

File: src/cli/commands/__tests__/lint.test.ts [hunk 1]
```diff
@@ -1,448 +1,172 @@
...
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain('5:10  warning  info-matches-stoplight  Info must contain Stoplight');
-          expect(ctx.stdout).not.toContain('Info object should contain `contact` object');
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('shows help when invalid arguments are passed', async () => {
+    const output = await run('lint --foo');
+    expect(output).toContain('lint a JSON/YAML document from a file or URL');
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: hm.. assuming this indeed is help for invalid arguments. If something relating to `foo` is printed, can we have the expectation with `foo` in it? If not, then it OK I guess :)

File: src/cli/commands/__tests__/lint.test.ts [hunk 2]
```diff
@@ -1,448 +1,172 @@
...
-        .it('given valid remote ruleset file, outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
-    });
+  it('calls lint with document and default options', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: `calls lint with document and default options`, so there's `doc`, so I know this is the document. But not sure what default options are? If default options is the object containing `.encoding` and `.format`, then what is `undefined`? @philsturgeon, do you think that we can use vars (like you did for `doc`) to name these things?

File: src/cli/commands/__tests__/lint.test.ts [hunk 3]
```diff
@@ -1,448 +1,172 @@
...
-            '3:6  warning  info-description  OpenAPI object info `description` must be present and non-empty string',
-          );
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('calls lint with document and custom encoding', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint --encoding utf16 ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: ditto

File: src/cli/commands/__tests__/lint.test.ts [hunk 4]
```diff
@@ -1,448 +1,172 @@
...
-          expect(ctx.stdout).toContain('22 problems (0 errors, 22 warnings, 0 infos)');
-          expect(ctx.stdout).toContain('OpenAPI 2.0 (Swagger) detected');
-        });
-    });
+  it('calls lint with document and custom encoding and format', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint -f json --encoding utf16 ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: ditto

File: src/cli/commands/__tests__/lint.test.ts [hunk 5]
```diff
@@ -1,448 +1,172 @@
...
-      .stdout()
-      .command(['lint', invalidOas3SpecPath, '--quiet'])
-      .it('does not log any additional feedback', ctx => {
-        expect(ctx.stdout).not.toContain('OpenAPI 3.x detected');
-      });
+  it('calls lint with document and custom rulesets', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    const ruleset = 'custom-ruleset.json';
+    await run(`lint -r ${ruleset} ${doc}`);
+    expect(lint).toBeCalledWith(doc, expect.any(Object), ruleset);
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: ok, so now I understand that `undefined` was just a `not defined ruleset`. But maybe it would read better if we did:
```
const ruleset = undefined;
```
in `calls lint with document and custom encoding`?

File: src/cli/commands/__tests__/lint.test.ts [hunk 6]
```diff
@@ -1,448 +1,172 @@
...
   });
 
-  describe('when --quiet flag is provided', () => {
-    test
-      .stdout()
-      .command(['lint', invalidOas3SpecPath, '--quiet'])
-      .it('does not log any additional feedback', ctx => {
-        expect(ctx.stdout).not.toContain('OpenAPI 3.x detected');
-      });
+  it('calls lint with document and custom rulesets', async () => {
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]:
```suggestion
  it('calls lint with document and custom ruleset', async () => {
```
later we have `calls lint with document and multiple custom rulesets`

File: src/cli/commands/__tests__/lint.test.ts [hunk 7]
```diff
@@ -1,448 +1,172 @@
...
-        api.get('/openapi').replyWithFile(200, invalidOas3SpecPath, {
-          'Content-Type': 'application/yaml',
-        }),
-      )
-      .stdout()
-      .command(['lint', 'http://foo.local/openapi'])
-      .it('outputs warnings in default format', ctx => {
-        expect(ctx.stdout).toContain('Info object should contain `contact` object');
-      });
+  it('writes formatted output to a file', async () => {
```
Comment (lag-of-death) [inline src/cli/commands/__tests__/lint.test.ts]: hm.. is this a duplicate?

File: src/cli/commands/lint.ts [hunk 8]
```diff
@@ -1,261 +1,92 @@
...
+const lintCommand: CommandModule = {
+  describe: 'lint a JSON/YAML document from a file or URL',
+  command: 'lint <document>',
+  builder: yargs =>
+    yargs
+      .positional('document', {
+        description: 'Location of a JSON/YAML document. Can be either a file or a fetchable resource on the web.',
+        type: 'string',
+      })
+      .fail(() => {
```
Comment (lag-of-death) [inline src/cli/commands/lint.ts]:
```suggestion
      .fail(showHelp)
```

File: src/cli/services/__tests__/linter.test.ts [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+              line: 5,
+            },
+          },
+          source: expect.stringContaining('/__tests__/__fixtures__/gh-474/common.yaml'),
+        }),
+      ]),
+    );
+  });
+
+  it('should handle relative path to a document #2', async () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]: do we need 2 of `should handle relative path to a document`?

File: src/cli/services/__tests__/linter.test.ts [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+        expect(output).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({ code: 'api-servers' }),
+            expect.objectContaining({ code: 'info-contact' }),
+          ]),
+        );
+      });
+
+      describe('and --skip-rule=info-contact is set', () => {
+        it('output other warnings but not info-contact', async () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]:
```suggestion
        it('outputs other warnings but not info-contact', async () => {
```

File: src/cli/services/__tests__/linter.test.ts [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+          expect(output).toEqual(expect.not.arrayContaining([expect.objectContaining({ code: 'info-contact' })]));
+          expect(output).toEqual(expect.not.arrayContaining([expect.objectContaining({ code: 'api-servers' })]));
+        });
+      });
+    });
+  });
+
+  describe('--ruleset', () => {
+    describe('extends feature', () => {
+      it('should extend a valid relative ruleset', () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]: hm.. I don't know. Some `it`s are in the form of `does` and others and in the form of `should do`. Maybe it would be good to unify these?

File: src/cli/services/__tests__/linter.test.ts [hunk 12]
```diff
@@ -0,0 +1,386 @@
...
+        return expect(run(`lint ${validCustomOas3SpecPath} -r ${validNestedRulesetPath}`)).resolves.toEqual([]);
+      });
+
+      it('should fail trying to extend an invalid relative ruleset', () => {
+        return expect(run(`lint ${validCustomOas3SpecPath} -r ${invalidNestedRulesetPath}`)).rejects.toThrowError(
+          ValidationError,
+        );
+      });
+
+      it('given remote nested ruleset should resolve', () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]: `given` in `it`? Not sure here, up to you :)

File: src/cli/services/__tests__/linter.test.ts [hunk 13]
```diff
@@ -0,0 +1,386 @@
...
+          });
+
+        return expect(run(`lint ${validCustomOas3SpecPath} -r http://foo.local/ruleset-master.yaml`)).resolves.toEqual(
+          [],
+        );
+      });
+    });
+
+    describe('when multiple ruleset options provided', () => {
+      it('given one is valid other is not, outputs "invalid ruleset" error', () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]: ditto

File: src/cli/services/__tests__/linter.test.ts [hunk 14]
```diff
@@ -0,0 +1,386 @@
...
+        .persist()
+        .get('/openapi')
+        .replyWithFile(200, validOas3SpecPath, {
+          'Content-Type': 'application/yaml',
+        });
+
+      return expect(run('lint http://foo.local/openapi')).resolves.toEqual([]);
+    });
+
+    it('throws if cannot load URI', () => {
```
Comment (lag-of-death) [inline src/cli/services/__tests__/linter.test.ts]: condition in `it`, not sure here

File: src/cli/services/linter.ts [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+export async function loadRulesets(cwd: string, rulesetFiles: string | string[] | null): Promise<IRuleset> {
+  if (rulesetFiles === null || rulesetFiles.length === 0) {
+    return {
+      functions: {},
+      rules: {},
+    };
+  }
+
+  return readRuleset(
+    (Array.isArray(rulesetFiles) ? rulesetFiles : [rulesetFiles]).map(
```
Comment (lag-of-death) [inline src/cli/services/linter.ts]: nice, me likey :+1:

File: src/cli/services/linter.ts [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+    ),
+  );
+}
+
+export async function lint(name: string, flags: ILintConfig, rulesetFile: Optional<string | string[]>) {
+  if (flags.verbose) {
+    console.info(`Linting ${name}`);
+  }
+
+  let targetUri = name;
```
Comment (lag-of-death) [inline src/cli/services/linter.ts]:
```suggestion
  const targetUri = (!/^https?:\/\//.test(name) && !isAbsolute(name)) ? resolve(process.cwd(), name) : name;
```

File: src/cli/services/linter.ts [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  if (invalidRules.length !== 0) {
+    if (!flags.quiet) {
+      console.warn(`ignoring invalid ${invalidRules.length > 1 ? 'rules' : 'rule'} "${invalidRules.join(', ')}"`);
+    }
+  }
+
+  if (skippedRules.length !== 0 && flags.verbose) {
+    if (flags.verbose) {
```
Comment (lag-of-death) [inline src/cli/services/linter.ts]: seems like this was already checked for in the line above

File: src/cli/commands/__tests__/lint.test.ts (hunk 1 above)
Comment (philsturgeon) [reply src/cli/commands/__tests__/lint.test.ts]: Help shows up whenever there is invalid arguments, so here are two issues:

1. the lack of the required `{document}` argument
2. the --foo option is nonsense

Either one of these two things would cause the help menu to display, but it will not complain about --foo.

Comment (philsturgeon) [reply src/cli/commands/__tests__/lint.test.ts]: Actually it probably should. This is copied from Prism, and I thought prism was only doing it when the argument was missing, not invalid options too.

- `spectral lint` should show help
- `spectral lint --foo` should complain about `--foo`

File: src/cli/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (philsturgeon) [reply src/cli/commands/__tests__/lint.test.ts]: I dunno @P0lip wrote these tests :D

File: src/types/config.ts [hunk 18]
```diff
@@ -7,8 +7,8 @@ export interface ILintConfig {
   encoding: string;
   format: OutputFormat;
   output?: string;
-  ruleset?: string[];
-  skipRule?: string[];
+  ruleset?: string | string[];
+  skipRule?: string | string[];
```
Comment (XVincentX) [inline src/types/config.ts]: Would it be worth to make this always an array so you do not have to deal with this?

File: src/cli/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: These tests have been ported :D. Eight way, the defaults in this particular case means no options specified. Its up to the command definition what those defaults are.

File: src/cli/services/linter.ts (hunk 17 above)
Comment (XVincentX) [inline src/cli/services/linter.ts]: Interesting stuff here — in theory this could be handled by a logger that will decide whether output something or not based on its configuration; so you can always output stuff and not having to use ifs everywhere. Could be a good next step

File: src/cli/services/linter.ts [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+import { httpAndFileResolver } from '../../resolvers/http-and-file';
+import { getDefaultRulesetFile } from '../../rulesets/loader';
+import { isOpenApiv2, isOpenApiv3 } from '../../rulesets/lookups';
+import { readRuleset } from '../../rulesets/reader';
+import { Spectral } from '../../spectral';
+import { IParsedResult, RuleCollection } from '../../types';
+import { ILintConfig } from '../../types/config';
+import { IRuleset } from '../../types/ruleset';
+
+export async function loadRulesets(cwd: string, rulesetFiles: string | string[] | null): Promise<IRuleset> {
```
Comment (XVincentX) [inline src/cli/services/linter.ts]: Looking at this code it's clear that if `rulesetFiles` is empty or null — nothing will effectively happen. 

You can make this function type-safe (and avoid the if and returning empty stuff) by declaring that the array must not be empty

```suggestion
export async function loadRulesets(cwd: string, rulesetFiles: string[] & { 0: string }): Promise<IRuleset> {
```

File: src/cli/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: In general, all options are passed. I'll explicitly pass them here rather than just a subset of them.

File: src/cli/commands/__tests__/lint.test.ts (hunk 3 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: In this case, we don't really care about other options - we just want to test whether this one is changed or not.

File: src/cli/services/linter.ts (hunk 17 above)
Comment (P0lip) [reply src/cli/services/linter.ts]: > y a logger that will decide whether output something or not based on its configuration; 

Yeah, I have a basic implementation of Logger already. Going to ship it in 4.2

File: src/cli/services/__tests__/linter.test.ts [hunk 20]
```diff
@@ -0,0 +1,386 @@
...
+const invalidNestedRulesetPath = resolve(__dirname, '__fixtures__/ruleset-extends-invalid.yaml');
+const standardOas3RulesetPath = resolve(__dirname, '../../../rulesets/oas3/index.json');
+const standardOas2RulesetPath = resolve(__dirname, '../../../rulesets/oas2/index.json');
+const draftRefSpec = resolve(__dirname, './__fixtures__/draft-ref.oas2.json');
+const draftNestedRefSpec = resolve(__dirname, './__fixtures__/draft-nested-ref.oas2.json');
+const validOas3SpecPath = resolve(__dirname, './__fixtures__/openapi-3.0-valid.yaml');
+const invalidOas3SpecPath = resolve(__dirname, '__fixtures__/openapi-3.0-no-contact.yaml');
+
+function run(command: string) {
+  const parser = yargs.command(lintCommand).help();
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/services/__tests__/linter.test.ts]: Let's see.

File: src/cli/commands/__tests__/lint.test.ts (hunk 7 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: It's a dupe, good catch.

File: src/cli/commands/__tests__/lint.test.ts [hunk 21]
```diff
@@ -1,448 +1,172 @@
...
 
-  describe('when using default ruleset file', () => {
-    let spy: SpyInstance;
-    beforeAll(() => {
-      spy = jest.spyOn(process, 'cwd').mockReturnValue(resolve(__dirname, '__fixtures__'));
+  it('errors upon exception', async () => {
+    const error = new Error('Failure');
+    (lint as jest.Mock).mockReset();
+    (lint as jest.Mock).mockReturnValueOnce({
+      // could be mockRejectedValueOnce, but Node 8 does not like it (different ticking?), so here is the workaround
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: Returning a promise does not help. Basically the fn is never called.

File: package.json [hunk 22]
```diff
@@ -51,21 +41,19 @@
...
+    "prebuild.binary": "yarn build",
     "pretest.karma": "node ./scripts/generate-karma-fixtures.js",
     "schema.update": "yarn typescript-json-schema --id \"http://stoplight.io/schemas/rule.schema.json\" --required tsconfig.json IRule --out ./src/meta/rule.schema.json",
     "test.harness": "jest -c ./jest.harness.config.js",
     "test.karma": "karma start",
     "test.prod": "yarn build && yarn lint && yarn test.karma && yarn test --coverage --maxWorkers=2",
     "test.update": "yarn test --updateSnapshot",
     "test.watch": "yarn test --watch",
-    "test": "jest"
+    "test": "jest --silent"
```
[the remark this answers is not in the document]
Comment (P0lip) [reply package.json]: We have a few console.* and they cause unnecessary noise. Should be improved once Logger is in place.

File: src/cli/services/__tests__/linter.test.ts (hunk 9 above)
Comment (P0lip) [reply src/cli/services/__tests__/linter.test.ts]: Yeah, there are 2 types of paths there. Can you think of any other naming here? :D

File: src/cli/commands/lint.ts (hunk 8 above)
Comment (P0lip) [reply src/cli/commands/lint.ts]: Can't do unfortunately. Fail method take 3 arguments.

File: src/cli/services/linter.ts (hunk 19 above)
Comment (P0lip) [reply src/cli/services/linter.ts]: It's okay if it's empty, isn't it? I mean, if one wants to do so, why not.

File: src/cli/commands/__tests__/lint.test.ts (hunk 5 above)
Comment (P0lip) [reply src/cli/commands/__tests__/lint.test.ts]: Yeah, seems fair.

File: src/cli/services/__tests__/linter.test.ts (hunk 12 above)
Comment (P0lip) [reply src/cli/services/__tests__/linter.test.ts]: @philsturgeon thoughts? I don't really mind

File: src/cli/services/linter.ts (hunk 19 above)
Comment (lag-of-death) [reply src/cli/services/linter.ts]: if `rulesetFiles: string[]`, then you only have to worry about one data structure though

Comment (P0lip) [reply src/cli/services/linter.ts]: Yeah, it's mostly for other developers' convenience, since that function is accessible.
I can move that to `loadRuleset` method, don't have any preference, up to you.

**Comment by P0lip** [comment]
@philsturgeon @XVincentX @lag-of-death 
Could you have another pass folks?

File: src/cli/services/linter.ts (hunk 19 above)
Comment (lag-of-death) [reply src/cli/services/linter.ts]: heh, don't know what moving "that to loadRuleset" solves. If it is about limiting number of data structures, then yeah, let's go for it :)

File: src/cli/commands/lint.ts [hunk 23]
```diff
@@ -1,261 +1,92 @@
...
+        showHelp();
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/commands/lint.ts]: No, we just don't make use of signale, so didn't move that part.
